core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -87,16 +87,25 @@ protected ObjectUnderFileSystem(AlluxioURI uri) {
    * Information about a single object in object UFS.
    */
   protected class ObjectStatus {
-    final long mContentLength;
-    final long mLastModifiedTimeMs;
-    final String mName;
+    private final long mContentLength;
+    private final long mLastModifiedTimeMs;
+    private final String mName;
+
+    private static final long INVALID_CONTENT_LENGTH = -1L;
+    private static final long INVALID_MODIFIED_TIME = -1L;
 
     public ObjectStatus(String name, long contentLength, long lastModifiedTimeMs) {
       mContentLength = contentLength;
       mLastModifiedTimeMs = lastModifiedTimeMs;
       mName = name;
     }
 
+    public ObjectStatus(String name) {
+      mContentLength = INVALID_CONTENT_LENGTH;
+      mLastModifiedTimeMs = INVALID_MODIFIED_TIME;
```

Comment:
@calvinjia `contentLength` and `lastModified` are only called for files to populate `UfsFileStatus`. In that case, these are valid values.